qiskit_optimization/algorithms/qrao/semideterministic_rounding.py

```diff
@@ -0,0 +1,92 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2023.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Semideterministic rounding module"""
+from dataclasses import dataclass
+
+from typing import Optional
+
+import numpy as np
+
+from qiskit_optimization.algorithms import OptimizationResultStatus, SolutionSample
+
+from .rounding_common import (
+    RoundingScheme,
+    RoundingContext,
+    RoundingResult,
+)
+
+
+@dataclass
+class SemideterministicRoundingResult(RoundingResult):
+    """Result of semideterministic rounding"""
+
+
+class SemideterministicRounding(RoundingScheme):
+    """Semideterministic rounding scheme
+
+    This is referred to as "Pauli rounding" in
+    https://arxiv.org/abs/2111.03167v2.
+    """
+
+    def __init__(self, *, seed: Optional[int] = None):
+        """
+        Args:
+            seed: Seed for random number generator, which is used to resolve
+                expectation values near zero to either +1 or -1.
+        """
+        super().__init__()
+        self.rng = np.random.RandomState(seed)
```

Comment:
public instance var - similar comment to a public instance var rng elsewhere

Reply:
Do you need to make `rng` public?

`RandomState` is a legacy random state generator.
https://numpy.org/doc/stable/reference/random/legacy.html
Could you replace it with `np.random.default_rng`?